File: authorization/src/main/kotlin/com/example/authorization/repository/AuthoritiesRepository.java
package com.example.authorization.repository;

import com.example.authorization.entity.*;
import org.springframework.data.jpa.repository.*;
import org.springframework.stereotype.*;

import java.util.*;

@Repository
public interface AuthoritiesRepository extends JpaRepository<Authorities, Integer> {
    Optional<Authorities> findAuthoritiesByAuthority(String authority);
}

File: authorization/src/main/kotlin/com/example/authorization/repository/AuthoritiesRepository.kt
package com.example.authorization.repository

import com.example.authorization.entity.Authorities
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface AuthoritiesRepository : JpaRepository<Authorities, Int> {
    fun findAuthoritiesByAuthority(authority: String): Authorities?
}
